java-squid/src/main/java/org/sonar/java/syntaxtoken/FirstSyntaxTokenFinder.java

```diff
@@ -19,450 +19,38 @@
-  public void visitExpressionStatement(ExpressionStatementTree tree) {
-    scan(tree.expression());
-  }
-
-  @Override
-  public void visitAssignmentExpression(AssignmentExpressionTree tree) {
-    scan(tree.variable());
-  }
-
-  @Override
-  public void visitArrayAccessExpression(ArrayAccessExpressionTree tree) {
-    scan(tree.expression());
-  }
-
-  @Override
-  public void visitMethodInvocation(MethodInvocationTree tree) {
-    scan(tree.methodSelect());
-  }
-
-  @Override
-  public void visitIfStatement(IfStatementTree tree) {
-    firstSyntaxToken = tree.ifKeyword();
-  }
-
-  @Override
-  public void visitAssertStatement(AssertStatementTree tree) {
-    firstSyntaxToken = tree.assertKeyword();
-  }
-
-  @Override
-  public void visitSwitchStatement(SwitchStatementTree tree) {
-    firstSyntaxToken = tree.switchKeyword();
-  }
-
-  @Override
-  public void visitCaseLabel(CaseLabelTree tree) {
-    firstSyntaxToken = tree.caseOrDefaultKeyword();
-  }
-
-  @Override
-  public void visitWhileStatement(WhileStatementTree tree) {
-    firstSyntaxToken = tree.whileKeyword();
-  }
-
-  @Override
-  public void visitDoWhileStatement(DoWhileStatementTree tree) {
-    firstSyntaxToken = tree.doKeyword();
-  }
-
-  @Override
-  public void visitForStatement(ForStatementTree tree) {
-    firstSyntaxToken = tree.forKeyword();
-  }
-
-  @Override
-  public void visitForEachStatement(ForEachStatement tree) {
-    firstSyntaxToken = tree.forKeyword();
-  }
-
-  @Override
-  public void visitBreakStatement(BreakStatementTree tree) {
-    firstSyntaxToken = tree.breakKeyword();
-  }
-
-  @Override
-  public void visitContinueStatement(ContinueStatementTree tree) {
-    firstSyntaxToken = tree.continueKeyword();
-  }
-
-  @Override
-  public void visitReturnStatement(ReturnStatementTree tree) {
-    firstSyntaxToken = tree.returnKeyword();
-  }
-
-  @Override
-  public void visitThrowStatement(ThrowStatementTree tree) {
-    firstSyntaxToken = tree.throwKeyword();
-  }
-
-  @Override
-  public void visitSynchronizedStatement(SynchronizedStatementTree tree) {
-    firstSyntaxToken = tree.synchronizedKeyword();
-  }
-
-  @Override
-  public void visitTryStatement(TryStatementTree tree) {
-    firstSyntaxToken = tree.tryKeyword();
-  }
-
-  @Override
-  public void visitCatch(CatchTree tree) {
-    firstSyntaxToken = tree.catchKeyword();
-  }
-
-  @Override
-  public void visitMethod(MethodTree tree) {
-    SyntaxToken firstModifierToken = getFirstModifierToken(tree.modifiers());
-    if (firstModifierToken != null) {
-      firstSyntaxToken = firstModifierToken;
-    } else {
-      TypeParameters typeParameters = tree.typeParameters();
-      if (!typeParameters.isEmpty()) {
-        firstSyntaxToken = typeParameters.openBracketToken();
-      } else if (tree.returnType() != null) {
-        scan(tree.returnType());
-      } else {
-        firstSyntaxToken = tree.simpleName().identifierToken();
+    } else if (tree.is(Tree.Kind.INFERED_TYPE)) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This branch's code block is the same as the block for the branch on line 36. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1871)